### PR TITLE
Dump TagRouterRule

### PR DESCRIPTION
**Pull request by carryxyh**
Dump TagRouterRule since the TagRouterRule can be changed to `null` by ConfigCenter.
Since his Issue[1] does not give a stack trace, I am not sure that his problem is caused by this.

[1] https://github.com/apache/incubator-dubbo/issues/3533

### CODE REVIEW

**Comment by chickenlj** [comment]
LGTM, I think your change is critical to guarantee consistent behavior when rule changes. 
But what's the possible relation with the issue mentioned here.

**Comment by carryxyh** [comment]
When I saw his issue before, he did not give his exception stack. Later, the issuer added the stack. I looked at the code and found that this place also caused some problems, so I modified it. @chickenlj

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagRouter.java [hunk 1]
```diff
@@ -87,7 +87,9 @@ public URL getUrl() {
             return invokers;
         }
 
-        if (tagRouterRule == null || !tagRouterRule.isValid() || !tagRouterRule.isEnabled()) {
+        // since the rule can be changed by config center, we should dump one to use.
+        TagRouterRule dump = tagRouterRule;
```
Comment (ralf0131) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagRouter.java]: Shall we mark it as final? Also I don't quite understand why it is named as dump?
If it is just a copy, will a name called `tagRouterRuleCopy` be better?

Comment (chickenlj) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagRouter.java]: Agree, tagRouterRuleCopy or CopyOfXxx would be better.

Comment (carryxyh) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/tag/TagRouter.java]: For me, the semantics of `copy` or `refer` are good choices. So, I'd like chose `copy` in this case.

**Pull request merged**